### PR TITLE
updating snmp_notification to work on XR.

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

**Comment by tphoney** [comment]
puppetlabs.net:57777 root lab
Run options: -v -- --seed 250

# Running:

Node under test:
- name  - r7t8jx4f0nzu3dr
- type  - R-IOSXRV9000-CH
- image - 6.0.0.20I

TestSnmpNotification#test_snmp_notification = 37.03 s = .

Finished in 43.223250s, 0.0231 runs/s, 0.1619 assertions/s.

1 runs, 7 assertions, 0 failures, 0 errors, 0 skips
Coverage report generated for MiniTest to /Users/tp/workspace/git/EXR/cisco_devenv/cisco-network-node-utils/coverage. 2241 / 4429 LOC (50.6%) covered.

**Comment by tphoney** [comment]
@chburket can you have a look at this ?

File: tests/test_snmpnotification.rb [hunk 1]
```diff
@@ -47,27 +55,53 @@ def test_snmp_notification
...
+
+      bfd.enable = true
+      flash_insertion.enable = true
+      assert_equal(bfd.enable, true)
+      assert_equal(flash_insertion.enable, true)
+
+      # Setting back to default and re-checking
+      bfd.enable = false
+      flash_insertion.enable = false
+      assert_equal(bfd.enable, false)
```
Comment (chburket) [inline tests/test_snmpnotification.rb]: should we check `assert_equal(flash_insertion.enable, false)` as well?

Comment (chburket) [inline tests/test_snmpnotification.rb]: Also, it looks like the nexus case and XR case are almost identical code.  If you renamed the notifications to something generic (`trap1` and `trap2` for example) and just passed a different value to the constructors for nexus vs. xr, would you be able to git rid of the duplication?

File: lib/cisco_node_utils/cmd_ref/snmpnotification.yaml [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+    set_value: "<state> snmp-server enable traps <trap_name>"
+  ios_xr:
+    get_value: '/^snmp-server traps <trap_name> ?$/'
+    set_value: "<state> snmp-server traps <trap_name>"
 
 notifications:
-  get_command: "sh run snmp all | include 'snmp-server enable traps'"
-  get_value: '/.*snmp-server enable traps (.*)?$/'
-  multiple:
+  nexus:
```
Comment (chburket) [inline lib/cisco_node_utils/cmd_ref/snmpnotification.yaml:18]: Nit: I assume this could be shortened to:

```
  multiple:
  nexus:
    get_value: '/.*snmp-server enable traps (.*)?$/'
  ios_xr:
    get_value: '/.*snmp-server traps (.*)?$/'
```

**Comment by tphoney** [comment]
@chburket  changes made, thanks for the feedback :)
cisco-network-node-utils git:(snmp_notification) ✗ bundle exec ruby tests/test_snmpnotification.rb -v -- r7t8jx4f0nzu3dr.delivery.puppetlabs.net:57777 root lab
Run options: -v -- --seed 25214

# Running:

Node under test:
- name  - r7t8jx4f0nzu3dr
- type  - R-IOSXRV9000-CH
- image - 6.0.0.20I

TestSnmpNotification#test_snmp_notification = 34.85 s = .

Finished in 40.765011s, 0.0245 runs/s, 0.1962 assertions/s.

1 runs, 8 assertions, 0 failures, 0 errors, 0 skips
Coverage report generated for MiniTest to /Users/tp/workspace/git/EXR/cisco_devenv/cisco-network-node-utils/coverage. 2241 / 4429 LOC (50.6%) covered.

File: lib/cisco_node_utils/cmd_ref/snmpnotification.yaml (hunk 2 above)
Comment (tphoney) [inline lib/cisco_node_utils/cmd_ref/snmpnotification.yaml:18]: i did not want to change the nexus command, as it is out of scope for the pr. It would be an excellent enhancement in another pr

Comment (chburket) [inline lib/cisco_node_utils/cmd_ref/snmpnotification.yaml:18]: My suggestion was just to move the duplicated `multiple` key outside of nexus/ios_xr, no other changes.  But it was a nit, so no problem if you don't want to mess with it.

**Comment by chburket** [comment]
:+1:

**Comment by tphoney** [comment]
@chburket thanks for the +1, who is next on the approval chain ?

File: lib/cisco_node_utils/cmd_ref/snmpnotification.yaml (hunk 2 above)
Comment (robert-w-gries) [inline lib/cisco_node_utils/cmd_ref/snmpnotification.yaml:18]: Keeping `multiple` as a top level key won't affect nexus so it should be fine to change.

**Comment by robert-w-gries** [comment]
@tphoney Do you have a nexus node to verify the PR against?  If so, please provide the results of the test for nexus.

**Comment by tphoney** [comment]
i do not have a readily available node to work with. apologies. @robert-w-gries

**Comment by robert-w-gries** [comment]
:+1: LGTM. Tested it against our own n9k and it passed:

``` bash
Node under test:
  - name  - n9k-108
  - type  - N9K-C9396PX
  - image - bootflash:///nxos.7.0.3.I3.1.bin

TestSnmpNotificationReceiver#test_create_destroy_multiple = 8.41 s = .
TestSnmpNotificationReceiver#test_create_destroy_single = 3.18 s = .
TestSnmpNotificationReceiver#test_create_destroy_single_ipv6 = 3.33 s = .
```

**Comment by tphoney** [comment]
Whos next on the merge chain :) ?

**Comment by robert-w-gries** [comment]
You got all the necessary approvals, so you can merge if ready.

**Comment by tphoney** [comment]
@robert-w-gries it is normally bad github etiquette to merge your own pr. can someone at cisco merge this. @chburket @mikewiebe

**Comment by tphoney** [comment]
much appreciated !!
